File: solutions/zig/rna-transcription/2/rna_transcription.zig
const std = @import("std");
const mem = std.mem;

pub fn toRna(allocator: mem.Allocator, dna: []const u8) mem.Allocator.Error![]const u8 {
    var rna = std.ArrayList(u8).init(allocator);
    for (dna) |nucleotide| {
        try rna.append(switch (nucleotide) {
            'G' => 'C',
            'C' => 'G',
            'T' => 'A',
            'A' => 'U',
            else => unreachable,
        });
    }
    return rna.toOwnedSlice();
}

File: solutions/ocaml/rna-transcription/2/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna (strand : dna list) : rna list =
  List.map
    (fun x -> match x with `G -> `C | `C -> `G | `T -> `A | `A -> `U)
    strand
